From a9083bb6f5450e0a705c8735ed9b460b3185266b Mon Sep 17 00:00:00 2001
From: Valentin Churavy <v.churavy@gmail.com>
Date: Mon, 2 May 2022 12:24:20 -0400
Subject: [PATCH] [Sanitizers] Fix compilation on musl 32bit

Without this compiling the santizers on i686-musl fail

Differential Revision: https://reviews.llvm.org/D124779
---
 .../lib/sanitizer_common/sanitizer_platform_limits_posix.h      | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.h b/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.h
index 4472b6efa963..e9213d2a905f 100644
--- a/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.h
+++ b/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.h
@@ -478,7 +478,7 @@ struct __sanitizer_dirent {
   unsigned short d_reclen;
   // more fields that we don't care about
 };
-#  elif SANITIZER_ANDROID || defined(__x86_64__) || defined(__hexagon__)
+#  elif SANITIZER_ANDROID || defined(__x86_64__) || defined(__hexagon__) || !defined(__GLIBC__)
 struct __sanitizer_dirent {
   unsigned long long d_ino;
   unsigned long long d_off;
-- 
2.35.1

